controller/controller_test.go

```diff
@@ -701,3 +702,42 @@ func TestDeleteRecyclesIP(t *testing.T) {
 		t.Fatal("svc2 didn't get an IP")
 	}
 }
+
+func TestRequestedIPs(t *testing.T) {
+	var tests = []struct {
+		testname     string
+		requestedIPs string
+		lbIP, lbIP2  net.IP
+		errContains  string
+	}{
+		{"Empty string", "", nil, nil, ""},
+		{"Ipv4 first", "10.0.0.0,  1000::", net.ParseIP("10.0.0.0"), net.ParseIP("1000::"), ""},
+		{"Ipv6 first", "1000::,10.0.0.0", net.ParseIP("1000::"), net.ParseIP("10.0.0.0"), ""},
+
+		{"Just one", "10.0.0.0", nil, nil, "Must be two addresses"},
+		{"More than two", "10.0.0.0, 1000::, 2000::", nil, nil, "Must be two addresses"},
+		{"Invalid 1", "10.0.0.400, 1000::", nil, nil, "Invalid address"},
+		{"Invalid 2", "10.0.0.0, 1000::2::2", nil, nil, "Invalid address"},
+		{"Same family", "1000::1, 1000::2", nil, nil, "Same family"},
```

Comment:
is having mcast ipv4 or ipv6 is valid IP address ? the validation check doesn't fail mcast 
```
{"Invalid 3", "10.0.0.0, ff00::2", nil, nil, "Invalid address"},
{"Invalid 4", "224.0.0.40, 1000::", nil, nil, "Invalid address"},
```

Reply:
can u add dual-stack service test case to `TestControllerMutation`